repositories/decision_phantoms_repository.go

```diff
@@ -0,0 +1,144 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/models/ast"
+	"github.com/checkmarble/marble-backend/repositories/dbmodels"
+	"github.com/checkmarble/marble-backend/utils"
+	"github.com/cockroachdb/errors"
+	"github.com/google/uuid"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type DecisionPhantomUsecaseRepository interface {
+	StorePhantomDecision(
+		ctx context.Context,
+		exec Executor,
+		decision models.DecisionWithRuleExecutions,
+		organizationId string,
+		testRunId string,
+		newDecisionId string) error
+
+	GetTestRunIterationByScenarioId(ctx context.Context, exec Executor, scenarioID string) (models.ScenarioIteration, error)
+}
+
+func (repo *MarbleDbRepository) StorePhantomDecision(
```

Comment:
yes we could.. but since I considered the testrun feature as a separate implementation, I thought we could do the same here.. moreover, with this approach we could easily support an evolution without impacting the decision design.